src/Parser/TIFFParser.js

```diff
@@ -0,0 +1,67 @@
+import * as UTIF from 'utif';
+import * as THREE from 'three';
+
+/**
+ * The TIFFParser module provides a [parse]{@link module:TIFFParser.parse}
+ * method that takes a TIFF in and gives a `THREE.DataTexture` that can be
+ * displayed in the view.
+ *
+ * It uses the [utif](https://www.npmjs.com/package/utif) library to parse the
+ * TIFF.
+ *
+ * @example
+ * Fetcher.arrayBuffer('image.tiff')
+ *     .then(TIFFParser.parse)
+ *     .then(function _(texture) {
+ *         var source = new FileSource({ parsedData: texture });
+ *         var layer = new ColorLayer('tiff', { source });
+ *         view.addLayer(layer);
+ *     });
+ *
+ * @module TIFFParser
+ */
+export default {
+    /**
+     * Parse a TIFF file and return a `THREE.DataTexture`.
+     *
+     * @param {ArrayBuffer} data - The TIFF file content to parse.
+     *
+     * @return {Promise} A promise resolving with a `THREE.DataTexture`.
+     */
+    parse(data) {
+        const IFD = UTIF.decode(data)[0];
+        UTIF.decodeImage(data, IFD);
+
+        // A 4096 limitation sounds the most probable here
+        // https://webglstats.com/webgl/parameter/MAX_TEXTURE_SIZE
+        if (IFD.width > 4096 || IFD.height > 4069) {
+            throw new Error('The loaded TIFF is too big: please reduce it to a maximum size of 4096 by 4096');
+        }
+
+        // Round to next power of two
+        const width = 2 ** (~~Math.log2(IFD.width - 1) + 1);
+        const height = 2 ** (~~Math.log2(IFD.height - 1) + 1);
+
+        // Avoid using alpha layer if the TIFF is already at the right size
+        const alpha = (IFD.width == width && IFD.height == height) ? 3 : 4;
+
+        const resizedData = new IFD.data.constructor(width * height * alpha);
+
+        const it = IFD.data.values();
+        const rowSize = IFD.height * alpha;
+        let rowOffset = 0;
+        for (let i = IFD.width; i > 0; i--) {
+            rowOffset = i * width * alpha;
+            resizedData.fill(255, rowOffset, rowOffset + rowSize);
```

Comment:
why you fill the `resizedData`.

Reply:
For the opacity, see other comment

Reply:
Verify if the tiff supports opacity. if not, could you use RGB texture?